### PR TITLE
Acmesmith Ruby Client

### PR DESCRIPTION
**Pull request by bitsapien**
MR relates to Issue #19 
@sorah , waiting for your review.

### CODE REVIEW

File: lib/acmesmith/client.rb [hunk 1]
```diff
@@ -0,0 +1,244 @@
+# require 'acmesmith/config'
```
Comment (sorah) [inline lib/acmesmith/client.rb]: Remove this line

File: lib/acmesmith/command.rb [hunk 2]
```diff
@@ -174,103 +70,41 @@ def show_private_key(common_name)
...
-
-    def account_key
-      @account_key ||= storage.get_account_key.tap do |x|
-        x.key_passphrase = account_key_passphrase if account_key_passphrase
-      end
-    end
-
-    def acme
-      @acme ||= Acme::Client.new(private_key: account_key.private_key, endpoint: config['endpoint'])
+    def acmesmith_client
```
Comment (sorah) [inline lib/acmesmith/command.rb]: Saying just `client` is enough?

File: spec/acmesmith_command_spec.rb [hunk 3]
```diff
@@ -1,11 +1,9 @@
-require 'acmesmith/config'
+# require 'acmesmith/config'
```
Comment (sorah) [inline spec/acmesmith_command_spec.rb]: remove this line

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,2 +1,3 @@
 $LOAD_PATH.unshift File.expand_path('../../lib', __FILE__)
-require 'acmesmith/command'
+require 'acmesmith/config'
+RSpec::Expectations.configuration.on_potential_false_positives = :nothing
```
Comment (sorah) [inline spec/spec_helper.rb]: Why do you need this?

**Comment by bitsapien** [comment]
Please review.

File: lib/acmesmith/client.rb [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+      else
+        storage.list_certificates.sort
+      end
+    end
+
+    def current(common_name)
+      storage.get_current_certificate_version(common_name)
+    end
+
+    def show_certificate(common_name, version: 'current', type: 'text')
```
Comment (sorah) [inline lib/acmesmith/client.rb]: This doesn't `show_`.

`get_` ?

File: lib/acmesmith/client.rb [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+    end
+
+    def save_certificate(common_name, version: 'current', mode: '0600', output:)
+      cert = storage.get_certificate(common_name, version: version)
+      File.open(output, 'w', mode.to_i(8)) do |f|
+        f.puts(cert.fullchain)
+      end
+    end
+
+    def show_private_key(common_name, version: 'current')
```
Comment (sorah) [inline lib/acmesmith/client.rb]: This doesn't `show_`.

`get_` ?

**Comment by bitsapien** [comment]
Made changes, and fixed a few errors. Please see.

**Comment by bitsapien** [comment]
Awaiting review.

**Pull request merged**